src/python/pants/backend/python/rules/pex.py

```diff
@@ -369,6 +374,12 @@ async def create_pex(
     # (execution_platform_constraint, target_platform_constraint) of this dictionary is "The output of
     # this command is intended for `target_platform_constraint` iff it is run on `execution_platform
     # constraint`".
+    description = request.description
+    if description is None:
+        if request.requirements.requirements:
```

Comment:
`requirements.requirements` is so clunky. 

In https://github.com/pantsbuild/pants/pull/9521/commits/bf2434ac4108b10f07482252bb79c6f91c054c12, I discovered a new pattern of using `Collection` for result and request types that reduces a lot of boilerplate. I've been thinking about the merits of adding a similar type of sugar that uses `FrozenOrderedSet` rather than `Collection` somehow. The result would be being able to simply say `for req in request.requirements`.